eventsources/sources/kafka/start.go

```diff
@@ -72,6 +76,100 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 	log.Info("start kafka event source...")
 	kafkaEventSource := &el.KafkaEventSource
 
+	if kafkaEventSource.ConsumerGroup == nil {
+		return el.partitionConsumer(ctx, log, kafkaEventSource, dispatch)
+	} else {
+		return el.consumerGroupConsumer(ctx, log, kafkaEventSource, dispatch)
+	}
+}
+
+func (listener *EventListener) consumerGroupConsumer(ctx context.Context, log *zap.SugaredLogger, kafkaEventSource *v1alpha1.KafkaEventSource, dispatch func([]byte) error) error {
+	config := sarama.NewConfig()
+
+	version, err := sarama.ParseKafkaVersion(kafkaEventSource.ConsumerGroup.KafkaVersion)
+	if err != nil {
+		log.Errorf("Error parsing Kafka version: %v", err)
+		return err
+	}
+	config.Version = version
+
+	switch kafkaEventSource.ConsumerGroup.RebalanceStrategy {
+	case "sticky":
+		config.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategySticky
+	case "roundrobin":
+		config.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRoundRobin
+	case "range":
+		config.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRange
+	default:
+		log.Info("Invalid rebalance strategy using default: range")
+		config.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRange
+	}
+
+	if kafkaEventSource.TLS != nil {
+		tlsConfig, err := common.GetTLSConfig(kafkaEventSource.TLS.CACertPath, kafkaEventSource.TLS.ClientCertPath, kafkaEventSource.TLS.ClientKeyPath)
+		if err != nil {
+			return errors.Wrap(err, "failed to get the tls configuration")
+		}
+		config.Net.TLS.Config = tlsConfig
+		config.Net.TLS.Enable = true
+	}
+
+	consumer := Consumer{
+		ready:            make(chan bool),
+		dispatch:         dispatch,
+		logger:           log,
+		kafkaEventSource: kafkaEventSource,
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	client, err := sarama.NewConsumerGroup([]string{kafkaEventSource.URL}, kafkaEventSource.ConsumerGroup.GroupName, config)
+	if err != nil {
+		log.Errorf("Error creating consumer group client: %v", err)
+		cancel()
+		return err
+	}
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			if err := client.Consume(ctx, []string{kafkaEventSource.Topic}, &consumer); err != nil {
+				log.Panicf("Error from consumer: %v", err)
```

Comment:
We support multiple events configured in one EventSource, `Panicf()` exits everything, it is not expected to do that because the other "good" events will be stopped.

Reply:
Good to know